src/v/pandaproxy/json/requests/fetch.h

```diff
@@ -42,22 +42,44 @@ class rjson_serialize_impl<model::record> {
       , _tpv(tpv)
       , _base_offset(base_offset) {}
 
-    void
+    bool
     operator()(::json::Writer<::json::StringBuffer>& w, model::record record) {
+        auto offset = _base_offset() + record.offset_delta();
+
         w.StartObject();
         w.Key("topic");
         ::json::rjson_serialize(w, _tpv.topic);
         w.Key("key");
-        rjson_serialize_fmt(_fmt)(w, record.release_key());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_key())) {
+            throw serialize_error(fmt::format(
+              "Unable to serialize key of record at offset {} in "
+              "topic:partition {}:{}",
+              offset,
+              _tpv.topic(),
+              _tpv.partition()));
```

Comment:
Should we return false instead of throw? IMO, the callsites are where the throw should happen